lib/wikidata/diff/analyzer.rb

```diff
@@ -128,6 +141,118 @@ def self.accumulate_totals(diff, total)
+    puts parsed_contents.length
+    # merge if not nil
+    if parsed_contents
+      parent_ids = []
+      revision_contents.merge!(parsed_contents) if parsed_contents
+      parsed_contents.values.each do |data|
+        parent_id = data[:parentid]
+        
+        if parent_id == 0 || parent_id.nil?
+          puts "parent_id is 0 or nil"
+          puts data[:revid]
+        else
+          parent_ids << parent_id
+        end
+      end
+      parent_contents_batch = get_revision_contents(parent_ids)
+      parent_contents.merge!(parent_contents_batch) if parent_contents_batch
+    end
+  end
+
+  result = {}
+  revision_contents.each do |revid, data|
+    parentid = data[:parentid]
+    parent_content = parent_contents[parentid] if parentid
+    current = data ? data[:content] : nil
+    parent = parent_content ? parent_content[:content] : nil
+    result[revid] = { current_content: current, parent_content: parent }
+  end
+  result
+end
+
+def self.get_revision_contents(revision_ids)
+  api_url = 'https://www.wikidata.org/w/api.php'
+  client = MediawikiApi::Client.new(api_url)
+
+  # remove duplicates if revision_ids exists
+  # check if duplicate revision_ids exist and print if exists
+
+  revision_ids = revision_ids.uniq if revision_ids
+
+  begin
+    response = client.action(
+      'query',
+      prop: 'revisions',
+      revids: revision_ids.join('|'),
+      rvslots: 'main',
+      rvprop: 'content|ids',
+      format: 'json'
+    )
+
+    if response.nil?
+      puts "No response received for revision IDs: #{revision_ids.join(', ')}"
+      return {}
+    end
+
+    parsed_contents = {}
+
+    # checks if it has pages
+    if response.data['pages'].nil?
+      puts "No pages found in the response for revision IDs: #{revision_ids.join(', ')}"
+      return nil
+    end
+
+    response.data['pages'].keys.each do |page|
+      page = response.data['pages'][page]
+      revisions = page['revisions']
+
+      revisions.each do |revision|
+        if revision['slots']['main']['contentmodel'] != 'wikibase-item'
+          puts "Content model is not wikibase-item"
+        else
+          content = revision['slots']['main']['*']
+          revid = revision['revid']
+          if revid == 0
+            parsed_contents[revid] = {content: nil, parentid: 0}
+          else
+            parentid = revision['parentid']
+            parsed_contents[revid] = {content: JSON.parse(content), parentid: parentid}
+          end
+        end
+      end
+    end
+    return parsed_contents
+  rescue MediawikiApi::ApiError => e
+    puts "Error retrieving revision content: #{e.message}"
+    return {}
+  rescue JSON::ParserError => e
+    puts "Error parsing JSON content: #{e.message}"
+    raise e
+  end
+end
+
+  
 # This method retrieves the content of a specific revision from the Wikidata API.
 # It takes a revision ID as input and returns the parsed content as a Ruby object.
   def self.get_revision_content(revision_id)
```

Comment:
It looks like this method is not longer used, since the new method can handle a single revision or many? If so, it should be deleted.

Reply:
@ragesoss Yeah as you have noticed `get_revision_content` and `get_parent_id` are not used anymore. I was using them to compare results with the new functions I wrote to see if they match. Will delete them eventually.